packages/search/src/search.tsx

```diff
@@ -228,6 +234,10 @@ const InnerSearch = (
 		}
 
 		setKeyword( '' );
+		if ( 'on-enter' === searchMode ) {
+			onSearch?.( '' );
+			onSearchChange?.( '' );
+		}
```

Comment:
Why trigger a search when the search box is being closed?

Reply:
Probably needs to be re-checked but I think this is so when you click the "X" clear/reset button the results fallback to default empty search.

Reply:
Confirming above